game-headed/src/main/java/org/triplea/game/client/parser/DefaultGameDetector.java

```diff
@@ -0,0 +1,28 @@
+package org.triplea.game.client.parser;
+
+import games.strategy.engine.framework.ui.GameChooserEntry;
+import games.strategy.engine.framework.ui.GameChooserModel;
+import java.util.Set;
+import java.util.function.Consumer;
+import javafx.concurrent.Service;
+import javafx.concurrent.Task;
+
+public class DefaultGameDetector extends Service<Set<GameChooserEntry>> implements GameDetector {
```

Comment:
Would it be possible to use composition instead of inheritance for the `Service`?

Reply:
Well this is just the way the service class works for JavaFX, not sure how to do that differently